File: mobile/factory_queue_mobile/lib/core/validation/contact_validation.dart
class ContactValidation {
  static const invalidEmailMessage = 'Geçerli bir e-posta adresi giriniz.';
  static const invalidPhoneMessage = 'Geçerli bir Türkiye cep telefonu numarası giriniz.';
  static const invalidPlateMessage = 'Geçerli bir Türkiye plakası giriniz.';
  static const missingContactMessage = 'E-posta veya telefon numarasından en az biri girilmelidir.';

  static final _emailRegex = RegExp(r'^[^@\s]+@[^@\s]+\.[^@\s]+$');

  static String? emptyToNull(String value) {
    final trimmed = value.trim();
    return trimmed.isEmpty ? null : trimmed;
  }

  static String? normalizeEmail(String? value) {
    final trimmed = value?.trim();
    return trimmed == null || trimmed.isEmpty ? null : trimmed.toLowerCase();
  }

  static String? normalizePhone(String? value) {
    final trimmed = value?.trim();
    if (trimmed == null || trimmed.isEmpty) {
      return null;
    }

    var normalized = trimmed.replaceAll(RegExp(r'[\s()-]'), '');
    if (normalized.startsWith('+90')) {
      return normalized;
    } else if (normalized.startsWith('90')) {
      return '+$normalized';
    } else if (normalized.length == 11 && normalized.startsWith('05')) {
      return '+90${normalized.substring(1)}';
    } else if (normalized.length == 10 && normalized.startsWith('5')) {
      return '+90$normalized';
    }

    return normalized;
  }

  static bool isValidEmail(String email) => _emailRegex.hasMatch(email);

  static bool isValidPhone(String phone) => RegExp(r'^\+905\d{9}$').hasMatch(phone);

  static String phoneInputValue(String? value) {
    final normalized = normalizePhone(value);
    return normalized != null && isValidPhone(normalized) ? normalized.substring(3) : value?.trim() ?? '';
  }

  static String? normalizePlate(String? value) {
    final trimmed = value?.trim();
    if (trimmed == null || trimmed.isEmpty) {
      return null;
    }

    return trimmed.toUpperCase();
  }

  static bool isValidPlate(String? plate) {
    if (plate == null) {
      return false;
    }

    final match = RegExp(r'^(\d{2})[A-ZÇĞİÖŞÜ]{1,3}\d{2,5}$').firstMatch(plate);
    if (match == null) {
      return false;
    }

    final province = int.tryParse(match.group(1)!);
    return province != null && province >= 1 && province <= 81;
  }
}

File: backend/FactoryQueueSystem.Api/Services/ContactValidation.cs
using System.Text.RegularExpressions;

namespace FactoryQueueSystem.Api.Services;

public static partial class ContactValidation
{
    public const string InvalidEmailMessage = "Geçerli bir e-posta adresi giriniz.";
    public const string InvalidPhoneMessage = "Geçerli bir Türkiye cep telefonu numarası giriniz.";
    public const string InvalidPlateMessage = "Geçerli bir Türkiye plakası giriniz.";
    public const string MissingContactMessage = "E-posta veya telefon numarasından en az biri girilmelidir.";

    public static string? NormalizeEmail(string? value) =>
        string.IsNullOrWhiteSpace(value) ? null : value.Trim().ToLowerInvariant();

    public static string? NormalizePhone(string? value)
    {
        if (string.IsNullOrWhiteSpace(value))
        {
            return null;
        }

        var normalized = value.Trim()
            .Replace(" ", string.Empty)
            .Replace("(", string.Empty)
            .Replace(")", string.Empty)
            .Replace("-", string.Empty);

        if (normalized.StartsWith("+90", StringComparison.Ordinal))
        {
            return normalized;
        }

        if (normalized.StartsWith("90", StringComparison.Ordinal))
        {
            return $"+{normalized}";
        }

        if (normalized.Length == 11 && normalized.StartsWith("05", StringComparison.Ordinal))
        {
            return $"+90{normalized[1..]}";
        }

        return normalized.Length == 10 && normalized.StartsWith('5')
            ? $"+90{normalized}"
            : normalized;
    }

    public static string? ToPhoneLocalPart(string? value)
    {
        var normalized = NormalizePhone(value);
        return normalized != null && IsValidPhone(normalized) ? normalized[3..] : normalized;
    }

    public static bool IsValidEmail(string email) => EmailRegex().IsMatch(email);

    public static bool IsValidPhone(string? phone) =>
        phone is { Length: 13 } &&
        phone.StartsWith("+905", StringComparison.Ordinal) &&
        phone[1..].All(char.IsDigit);

    public static string? NormalizePlate(string? value)
    {
        if (string.IsNullOrWhiteSpace(value))
        {
            return null;
        }

        return value.Trim().ToUpperInvariant();
    }

    public static bool IsValidPlate(string? plate)
    {
        if (plate == null)
        {
            return false;
        }

        var match = PlateRegex().Match(plate);
        if (!match.Success || !int.TryParse(match.Groups["province"].Value, out var province))
        {
            return false;
        }

        return province is >= 1 and <= 81;
    }

    [GeneratedRegex(@"^[^@\s]+@[^@\s]+\.[^@\s]+$", RegexOptions.CultureInvariant)]
    private static partial Regex EmailRegex();

    [GeneratedRegex(@"^(?<province>\d{2})[A-ZÇĞİÖŞÜ]{1,3}\d{2,5}$", RegexOptions.CultureInvariant)]
    private static partial Regex PlateRegex();
}
